### DISCUSSION TITLE
Error when ProvisioningUrl passed to otJoinerStart() is null

### DISCUSSION THREAD

**Discussion by OlivierGre** [openthread/openthread#7891] [Q&A] [answered]
[otjoinerstart documentation](https://openthread.io/reference/group/api-joiner#otjoinerstart) says the 'aProvisioningUrl' parameter can be null:

> [in] aProvisioningUrl | A pointer to the Provisioning URL (may be NULL).

but, it seems not accepted. In https://github.com/openthread/openthread/blob/2ce3d3bf0218566484be2e9943b95c755cefebe3/src/core/meshcop/joiner.cpp#L131
there is:

    VerifyOrExit(aProvisioningUrl == nullptr || IsValidUtf8String(aProvisioningUrl), error = kErrorInvalidArgs);
    VerifyOrExit(aVendorName == nullptr || IsValidUtf8String(aVendorName), error = kErrorInvalidArgs);
    VerifyOrExit(aVendorSwVersion == nullptr || IsValidUtf8String(aVendorSwVersion), error = kErrorInvalidArgs);

So I get an error OT_ERROR_INVALID_ARGS if the ProvisioningUrl is null. Is it an error in the code?

Thank you

**Comment by abtink** [comment, marked as answer]
@OlivierGre this line here does allow it to be `nullptr`. It verifies it to be either `nullptr` or a valid UTF8 (and exits if it is not).
```c++
VerifyOrExit(aProvisioningUrl == nullptr || IsValidUtf8String(aProvisioningUrl), error = kErrorInvalidArgs);
````

The `PrepareJoinerFinalizeMessage()` also seem to allow it to be `nullptr`.

May be the error is caused by something else.

**Comment by OlivierGre** [comment]
> @OlivierGre this line here does allow it to be `nullptr`. It verifies it to be either `nullptr` or a valid UTF8 (and exits if it is not).
> 
> ```c++
> VerifyOrExit(aProvisioningUrl == nullptr || IsValidUtf8String(aProvisioningUrl), error = kErrorInvalidArgs);
> ```
> 
> The `PrepareJoinerFinalizeMessage()` also seem to allow it to be `nullptr`.
> 
> May be the error is caused by something else.

You're right. I have found that the error comes from:

`    VerifyOrExit(IsPskdValid(aPskdString), error = kErrorInvalidArgs);`

and it is due to our pskd that is too small. It should be at least 6 characters long.

Thanks for your support.
